src/compiler.ts

```diff
@@ -7903,6 +7915,14 @@ export class Compiler extends DiagnosticEmitter {
       this.currentType = Type.bool;
       return this.module.unreachable();
     }
+    if (expression.expression.kind == NodeKind.IDENTIFIER) {
+      const identifier = <IdentifierExpression>expression.expression;
+      const element = flow.lookup(identifier.text);
+      if (element && element instanceof TypedElement) {
```

Comment:
`instanceof` is operator so extra check for null/undefined is unnecessary.
```suggestion
      if (element instanceof TypedElement) {
```

Reply:
Do you think this is a good idea to resolve type narrow?

Reply:
Yeah, definitely after `instanceof` guard LHS should handle as non-nullable